src/components/Trade/TradeInput.tsx

```diff
@@ -163,142 +164,171 @@ export const TradeInput = ({ history }: RouterProps) => {
-              getQuote({ amount, sellAsset, buyAsset, action })
-            }}
-            inputLeftElement={
-              <TokenButton
-                onClick={() => history.push('/trade/select/buy')}
-                logo={buyAsset?.currency?.icon}
-                symbol={buyAsset?.currency?.symbol}
+            </FormControl>
+            <FormControl
+              rounded=''
+              my={6}
+              pl={6}
+              pr={2}
+              display='flex'
+              alignItems='center'
+              justifyContent='space-between'
+            >
+              <IconButton
+                onClick={switchAssets}
+                aria-label='Switch'
+                isRound
+                icon={<FaArrowsAltV />}
+                isLoading={!quote || action || error ? true : false}
+                _loading={{ color: 'blue.500' }}
               />
-            }
-          />
-        </FormControl>
-        <Button
-          type='submit'
-          size='lg'
-          width='full'
-          colorScheme={error ? 'red' : 'blue'}
-          isLoading={isSubmitting || isSendMaxLoading || !!action}
-          isDisabled={!isDirty || !isValid || !!action || !wallet}
-          style={{
-            whiteSpace: 'normal',
-            wordWrap: 'break-word'
-          }}
-        >
-          <Text translation={!wallet ? 'common.connectWallet' : error ?? 'trade.previewTrade'} />
-        </Button>
+              <Box display='flex' alignItems='center' color='gray.500' fontSize='sm' spacing='24px'>
+                {!quote || action || error ? (
+                  <Text translation={error ? 'common.error' : 'trade.searchingRate'} />
+                ) : (
+                  <>
+                    <RawText whiteSpace={'pre'}>{`1 ${sellAsset.currency?.symbol} = `}</RawText>
+                    <NumberFormat
+                      value={firstNonZeroDecimal(bnOrZero(quote?.rate))}
+                      displayType={'text'}
+                      thousandSeparator={true}
+                    />
+                    <RawText whiteSpace={'pre'}>{` ${buyAsset?.currency?.symbol}`}</RawText>
+                    <HelperTooltip label={translate('trade.tooltip.rate')} />
+                    {/* <Button
+                      label={'Confirm'}
+                      onClick={async () =>
+                        history.push({
+                          pathname: '/trade/confirm',
+                          state: { fiatRate: await getGetFiatRate(true) }
+                        })
+                      }
+                    />
+                    <Button
+                      label={'Approval'}
+                      onClick={async () =>
+                        history.push({
+                          pathname: '/trade/approval',
+                          state: { fiatRate: await getGetFiatRate(true) }
+                        })
+                      }
+                    /> */}
```

Comment:
I'd prefer not to keep commented out code.

Reply:
I will remove this bit and amend the commit.